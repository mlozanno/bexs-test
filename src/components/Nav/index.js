import React from 'react';
import { Link } from 'react-router-dom';

const Nav = () => (
	<ul data-testid="nav">
		<li>
			<Link to="/">Home</Link>
		</li>

		<li>
			<Link to="/carteira">Carteira</Link>
		</li>

		<li>
			<Link to="/checkout">Checkout</Link>
		</li>
	</ul>
);

export default Nav;
