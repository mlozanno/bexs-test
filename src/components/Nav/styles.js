import styled from 'styled-components';
import { spacing, zIndex, colors } from '~/styles/theme';

export const StyledNav = styled.nav`
	@media (max-width: 767px) {
		position: fixed;
		top: 0;
		bottom: 0;
		left: 100%;
		display: flex;
		align-items: center;
		justify-content: center;
		width: 100%;
		height: 100vh;
		background-color: ${colors.primary};

		a {
			color: white;
		}

		&.opened {
			left: 0;
		}
	}
`;

export const ToggleButton = styled.button`
	margin: 0;
	padding: 0;
	background: transparent;
	border: none;

	&.close {
		position: absolute;
		top: ${spacing(2)};
		right: ${spacing(2)};
		z-index: ${zIndex.floated};
		color: ${colors.light};
	}

	@media (min-width: 768px) {
		display: none;
	}
`;

export const MenuList = styled.ul`
	@media (max-width: 767px) {
		text-align: center;

		> li + li {
			margin-top: ${spacing(2)};
		}
	}

	@media (min-width: 768px) {
		display: flex;

		> li + li {
			margin-left: ${spacing(2)};
		}
	}
`;
