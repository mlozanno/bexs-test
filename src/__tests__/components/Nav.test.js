import React from 'react';
import { MemoryRouter } from 'react-router';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Nav from '~/components/Nav';

describe('<Nav />', () => {
	it('should be a function', () => {
		expect(typeof Nav).toBe('function');
	});

	it('should be render component', () => {
		const { getByTestId } = render(
			<MemoryRouter>
				<Nav />
			</MemoryRouter>
		);

		const nav = getByTestId('nav');

		expect(nav).toMatchSnapshot();
	});
});
