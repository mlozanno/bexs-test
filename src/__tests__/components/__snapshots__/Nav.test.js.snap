// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should be render component 1`] = `
<ul
  class="sc-AxiKw kuWfqk"
  data-testid="nav"
>
  <li>
    <a
      href="/"
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/checkout"
    >
      Checkout
    </a>
  </li>
  <li>
    <a
      href="/carteira"
    >
      Carteira
    </a>
  </li>
</ul>
`;
